_examples/router-walk/main.go

```diff
@@ -0,0 +1,45 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi"
+)
+
+func main() {
+	r := chi.NewRouter()
+	r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("root."))
+	})
+
+	r.Route("/road", func(r chi.Router) {
+		r.Get("/left", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("left road"))
+		})
+		r.Post("/right", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("right road"))
+		})
+	})
+
+	r.Put("/ping", Ping)
+
+	methodCount := make(map[string]int)
+	walkFunc := func(method string, route string, handler http.Handler, middlewares ...func(http.Handler) http.Handler) error {
+		methodCount[method]++
+		return nil
```

Comment:
You might wanna print the routes in the following format instead?

```
GET /
GET /ping
```

Reply:
Yes, looks good!